### PR TITLE
B-20671 Parse TPPS Paid Invoice INT Part 2

### PR DESCRIPTION
**Pull request by traskowskycaci**
## Summary

The first round of tests written for this PR used example test files that were downloaded from the SFTP pickup folder and opened in vscode. However, those files were found to have a slightly different encoding while still directly inside the pickup folder.

Downloaded/opened in vscode:
image

While still in the SFTP pickup folder directly:
image

This new test parses the version of the file that is directly inside of the SFTP directory. I also fixed a few bugs in the parsing logic, specifically where the getPriceParts was not properly handing non-decimal values (as it should in the case of even dollar amounts with no change).

After this test, we now have three different ways of testing the parsing logic:
1. Parsing a string representation of the file.
2. Parsing and storing in the database a version of the file that has been downloaded from the SFTP pickup folder and viewed in vscode.
3. Parsing and storing in the database a version of the file that is directly from the SFTP pickup folder, with slightly different encoding than the previous step.

### How to test

1. Open the new `pkg/services/invoice/fixtures/tpps_paid_invoice_report_testfile_tpps_pickup_dir.csv` file in vscode and take a look at how it's formatted.
2. In `pkg/services/invoice/process_tpps_paid_invoice_report_test.go`, verify that the `successfully proccesses a TPPSPaidInvoiceReport from a file directly from the TPPS pickup directory and stores it in the database` test passes, as well as the previous parsing tests.

## Screenshots

This shows the SFTP pickup folder and the files inside.
image

### CODE REVIEW

**Comment by traskowskycaci** [comment]
Flaky server tests failing

File: pkg/services/invoice/process_tpps_paid_invoice_report_test.go [hunk 1]
```diff
@@ -144,6 +188,256 @@ func (suite *ProcessTPPSPaidInvoiceReportSuite) TestParsingTPPSPaidInvoiceReport
 		}
 	})
 
+	suite.Run("successfully proccesses a TPPSPaidInvoiceReport from a file directly from the TPPS pickup directory and stores it in the database", func() {
```
Comment (danieljordan-caci) [inline pkg/services/invoice/process_tpps_paid_invoice_report_test.go:191]: typo in `processes`

**Review by danieljordan-caci — CHANGES_REQUESTED** [review]
Itty bitty typo

File: pkg/services/invoice/process_tpps_paid_invoice_report_test.go (hunk 1 above)
Comment (traskowskycaci) [reply pkg/services/invoice/process_tpps_paid_invoice_report_test.go]: Fixed! [Resolution](https://github.com/transcom/mymove/pull/13527/commits/b8da51cfdcd61c1810f9798e80afff55aad694e5)

**Review by danieljordan-caci — APPROVED** [review]

**Comment by traskowskycaci** [comment]
seeing some conflicts, will resolve

**Comment by traskowskycaci** [comment]
Client test failed after merging in latest int, not marked as flaky in Circle but does pass for me locally

image

**Comment by traskowskycaci** [comment]
> Client test failed after merging in latest int, not marked as flaky in Circle but does pass for me locally
> 
> image

Just a flaky client test, still good to review

File: pkg/edi/tpps_paid_invoice_report/parser.go [hunk 2]
```diff
@@ -68,6 +68,7 @@ func ParseTPPSReportEntryForOneRow(row []string, columnIndexes map[string]int, h
 				}
 			}
 			processedEntry = strings.TrimSpace(processedEntry)
+			processedEntry = strings.TrimLeft(processedEntry, "�")
```
Comment (deandreJones) [inline pkg/edi/tpps_paid_invoice_report/parser.go:71]: is that supposed to be a weird ascii char?

**Review by cameroncaci — APPROVED** [review]
Good, just curious about what Dre found too https://github.com/transcom/mymove/pull/13527/files#r1729082481

File: pkg/edi/tpps_paid_invoice_report/parser.go (hunk 2 above)
Comment (traskowskycaci) [reply pkg/edi/tpps_paid_invoice_report/parser.go:71]: Yep! It starts off the version of the file that lives in the SFTP folder. 
image

@deandreJones @cameroncaci

Comment (deandreJones) [reply pkg/edi/tpps_paid_invoice_report/parser.go:71]: every file starts that way?

Comment (deandreJones) [reply pkg/edi/tpps_paid_invoice_report/parser.go:71]: well I see that - that's how it originally shows,, but will it be that way after the app grabs it and parses it

Comment (traskowskycaci) [reply pkg/edi/tpps_paid_invoice_report/parser.go:71]: Yep, every file currently in that folder does

image


image


image


image

Comment (traskowskycaci) [reply pkg/edi/tpps_paid_invoice_report/parser.go:71]: Talked with Dre on slack about his question - the parsing logic should still work even if there isn't the weird character to lead off the file. However, we won't truly know this parsing logic works until we can hook up to the final agreed on pickup dir - just writing tests for as many versions of the file I can predict at this point

**Pull request merged**